util/reggen/ip_block.py

```diff
@@ -261,12 +273,15 @@ def from_raw(param_defaults: List[Tuple[str, str]],
                              "by {})."
                              .format(name, dev_if_names,
                                      list(reg_block_names)))
+        attr = check_optional_str(rd.get('attr', None),
+                                  'attr field of ' + what)
```

Comment:
Can you add a check here that `attr` is one of the supported values?

Reply:
do we have a function I can use for check it? Thanks

Reply:
I don't think so: haven't you just added that field? Anyway, I was just imagining something like this:
```python3
if attr not in [None, 'templated', 'reggen_top', 'reggen_only']:
  raise ValueError('Unsupported value for attr field of {}: {!r}'.format(what, attr))
```